psydac/api/tests/test_api_2d_fields.py

```diff
@@ -162,15 +160,14 @@ def run_non_linear_poisson(filename, comm=None):
     l2norm_du_h  = discretize(l2norm_du , Omega_h, Vh)
 
     # First guess: zero solution
-    u_h  = FemField(Vh)
+    u_h = FemField(Vh)
 
     # Newton iteration
     for n in range(N):
 
         print()
         print('==== iteration {} ===='.format(n))
-        X = equation_h.solve(u=u_h)
-        du_h = FemField(Vh, X)
+        du_h = equation_h.solve(u=u_h)
```

Comment:
@campospinto What do you think of this? The result is a FemField now...

Reply:
That's nice indeed. More consistent with the fact that u is a FemField also.